### PR TITLE
Add GHC Filesystem 

### PR DESCRIPTION
**Pull request by kenshen112**
This is a PR to add GHC filesystem to pcsx2 common utilittes

### CODE REVIEW

**Comment by xTVaser** [comment]
Thoughts on putting this into `3rdparty` instead as it is a 3rdparty dependency?

**Comment by arcum42** [comment]
Yeah, given we didn't write it, 3rdparty would be a better location. While it is just one file, it's a third party file, and Utilities is mostly things we wrote. I can think of an exception or two, but then, maybe any exceptions should be moved, too...

**Comment by kenshen112** [comment]
Roger

**Comment by tellowkrinkle** [comment]
Thoughts on using a directory structure more like
```
3rdparty
└── filesystem
    ├── CMakeLists.txt
    ├── filesystem.vcxproj
    ├── filesystem.vcxproj.filters
    └── include
        └── ghc
            ├── filesystem.hpp
            └── fs_std.hpp
```

File: common/include/Utilities/filesystem.h [hunk 1]
```diff
@@ -0,0 +1,5689 @@
+//---------------------------------------------------------------------------------------
+//
+// ghc::filesystem - A C++17-like filesystem implementation for C++11/C++14/C++17
+//
+//---------------------------------------------------------------------------------------
```
Comment (tadanokojin) [inline common/include/Utilities/filesystem.h:5]: duplicate file

File: pcsx2/CMakeLists.txt [hunk 2]
```diff
@@ -945,7 +945,7 @@ set(pcsx2FinalSources
 set(pcsx2FinalLibs
     Utilities
     x86emitter
-    fmt::fmt
+	fmt::fmt
```
Comment (tadanokojin) [inline pcsx2/CMakeLists.txt:948]: Unnecessary formatting change

File: pcsx2/CMakeLists.txt [hunk 3]
```diff
@@ -978,7 +978,8 @@ endif()
 # additonal include directories
 include_directories(
     gui
-    x86
+	x86
+	../3rdparty/include
```
Comment (tadanokojin) [inline pcsx2/CMakeLists.txt:982]: Unnecessary formatting change

**Comment by TheLastRar** [comment]
What is the status of this pr?

**Review by xTVaser — CHANGES_REQUESTED** [review]
Can we get these commits squashed perhaps as follows:
- deps: Add ghc::filesystem header
- vs: Add new 3rdparty include folder to project properties
- cmake: Configure new 3rdparty include folder

Other than cleaning up the commits I don't see a reason for this to be held up.  
- It's just adding a new library that is unused (yet)
- CI is building successfully so project config changes were successful.
- And it's already being used successfully here https://github.com/PCSX2/pcsx2/pull/4005 so the new library should be usable by whoever needs it once merged.

**Pull request merged**